<img src="epl.jpg" width=450px>

### Introduction
Data used in this analysis was derived from GitHub via JSON API resquest. The primary source of the data is Openfootball. The request made was for the __2019/2020 English Premier League__ season matches dataset. 

The intend for this exercise was to access the dataset and determine by analysis information which are not immediately visible when viewing league standings table. Consequently, the aim is primarily not to reproduce the league standings table but instead, to show among other things miniscule and or granular details possible, such as follows: 

* the team that sufferred the worsre defeat and the team that inflicted it
* the team with the highest numbers of goals scored
* the highest number of goals scored
* the teams that had more goals scored away than at home
* the team that may have won the league
* the team with the highest number of draws 
* the overall goals for and against for each team during the season
* the team with the highest number losses at home and away from home
* the team with the highest number of wins at home and away
* number of losses
* number of draws
* number of wins
* the total numbers of goals scored during the season
* the average number of goals scored by each team
* the total numbers of goals scored by each team
* conceeded goals home and away
* the team that scored more goals away than at home
* top five contenders (UEFA & Europa Bound teams)
* last three contenders (relagation bound teams)




### Making the required request

In [2]:
import pandas as pd
import json
import requests

In [3]:
url="https://raw.githubusercontent.com/openfootball/football.json/master/2019-20/en.1.json"
headers={'Authorization': 'token fcbca88d83668f45abfdf7fa004e61542a21c9cf' }

In [4]:
response=requests.get(url,headers=headers)

In [5]:
data=response.json()

In [6]:
data

{'name': 'Premier League 2019/20',
 'matches': [{'round': 'Matchday 1',
   'date': '2019-08-09',
   'team1': 'Liverpool FC',
   'team2': 'Norwich City FC',
   'score': {'ft': [4, 1]}},
  {'round': 'Matchday 1',
   'date': '2019-08-10',
   'team1': 'West Ham United FC',
   'team2': 'Manchester City FC',
   'score': {'ft': [0, 5]}},
  {'round': 'Matchday 1',
   'date': '2019-08-10',
   'team1': 'Burnley FC',
   'team2': 'Southampton FC',
   'score': {'ft': [3, 0]}},
  {'round': 'Matchday 1',
   'date': '2019-08-10',
   'team1': 'AFC Bournemouth',
   'team2': 'Sheffield United FC',
   'score': {'ft': [1, 1]}},
  {'round': 'Matchday 1',
   'date': '2019-08-10',
   'team1': 'Crystal Palace FC',
   'team2': 'Everton FC',
   'score': {'ft': [0, 0]}},
  {'round': 'Matchday 1',
   'date': '2019-08-10',
   'team1': 'Watford FC',
   'team2': 'Brighton & Hove Albion FC',
   'score': {'ft': [0, 3]}},
  {'round': 'Matchday 1',
   'date': '2019-08-10',
   'team1': 'Tottenham Hotspur FC',
   'team2': 

In [172]:
match_date=[]
team_1=[]
team_2=[]
ft_score1=[]
ft_score2=[]
win_1=[]
win_2=[]
draw_1=[]
draw_2=[]
for m in data["matches"]:
    match_date.append(m["date"])
    team_1.append(m["team1"])
    team_2.append(m["team2"])
    ft_score1.append(m["score"]["ft"][0])
    ft_score2.append(m["score"]["ft"][1])
    win_1.append(m["score"]["ft"][0] > m["score"]["ft"][1])
    win_2.append(m["score"]["ft"][0] < m["score"]["ft"][1])
    draw_1.append(m["score"]["ft"][0] == m["score"]["ft"][1])
    draw_2.append(m["score"]["ft"][0] == m["score"]["ft"][1])

In [173]:
import pandas as pd
dframe=pd.DataFrame({'date':match_date,'team1':team_1,'score1':ft_score1,
                     'team2':team_2,'score2':ft_score2,'win1':win_1,
                     'win2':win_2,'draw1':draw_1,'draw2':draw_2})

In [174]:
dframe

date                team1  score1                 team2  score2  \
0    2019-08-09         Liverpool FC       4       Norwich City FC       1   
1    2019-08-10   West Ham United FC       0    Manchester City FC       5   
2    2019-08-10           Burnley FC       3        Southampton FC       0   
3    2019-08-10      AFC Bournemouth       1   Sheffield United FC       1   
4    2019-08-10    Crystal Palace FC       0            Everton FC       0   
..          ...                  ...     ...                   ...     ...   
375  2020-07-26    Leicester City FC       0  Manchester United FC       2   
376  2020-07-26   Manchester City FC       5       Norwich City FC       0   
377  2020-07-26  Newcastle United FC       1          Liverpool FC       3   
378  2020-07-26       Southampton FC       3   Sheffield United FC       1   
379  2020-07-26   West Ham United FC       1        Aston Villa FC       1   

      win1   win2  draw1  draw2  
0     True  False  False  False  
1    False   True  False  False  
2     True  False  False  False  
3    False  False   True   True  
4    False  False   True   True  
..     ...    ...    ...    ...  
375  False   True  False  False  
376   True  False  False  False  
377  False   True  False  False  
378   True  False  False  False  
379  False  False   True   True  

[380 rows x 9 columns]

In [196]:
dframe.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 380 entries, 0 to 379
Data columns (total 9 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   date    380 non-null    object
 1   team1   380 non-null    object
 2   score1  380 non-null    int64 
 3   team2   380 non-null    object
 4   score2  380 non-null    int64 
 5   win1    380 non-null    bool  
 6   win2    380 non-null    bool  
 7   draw1   380 non-null    bool  
 8   draw2   380 non-null    bool  
dtypes: bool(4), int64(2), object(3)
memory usage: 16.5+ KB


In [199]:
dframe.describe(include=['object','bool'])

date                 team1                 team2   win1   win2  \
count          380                   380                   380    380    380   
unique         115                    20                    20      2      2   
top     2020-07-26  Tottenham Hotspur FC  Tottenham Hotspur FC  False  False   
freq            10                    19                    19    208    264   

        draw1  draw2  
count     380    380  
unique      2      2  
top     False  False  
freq      288    288

In [200]:
dframe.describe()

score1      score2
count  380.000000  380.000000
mean     1.515789    1.205263
std      1.248646    1.200252
min      0.000000    0.000000
25%      1.000000    0.000000
50%      1.000000    1.000000
75%      2.000000    2.000000
max      8.000000    9.000000

In [178]:
dr_home=dframe.groupby(df['team1'])[['draw1']].sum()
dr_home

draw1
team1                            
AFC Bournemouth                 6
Arsenal FC                      6
Aston Villa FC                  3
Brighton & Hove Albion FC       7
Burnley FC                      4
Chelsea FC                      3
Crystal Palace FC               5
Everton FC                      7
Leicester City FC               4
Liverpool FC                    1
Manchester City FC              2
Manchester United FC            7
Newcastle United FC             8
Norwich City FC                 3
Sheffield United FC             3
Southampton FC                  3
Tottenham Hotspur FC            3
Watford FC                      6
West Ham United FC              4
Wolverhampton Wanderers FC      7

In [204]:
dr_away=dframe.groupby(df['team2'])[['draw2']].sum()
dr_away

draw2
team2                            
AFC Bournemouth                 1
Arsenal FC                      8
Aston Villa FC                  5
Brighton & Hove Albion FC       7
Burnley FC                      5
Chelsea FC                      3
Crystal Palace FC               5
Everton FC                      3
Leicester City FC               4
Liverpool FC                    2
Manchester City FC              1
Manchester United FC            5
Newcastle United FC             3
Norwich City FC                 3
Sheffield United FC             9
Southampton FC                  4
Tottenham Hotspur FC            8
Watford FC                      4
West Ham United FC              5
Wolverhampton Wanderers FC      7

In [179]:
dr=pd.concat([dr_home,dr_away],axis=1)
dr

draw1  draw2
AFC Bournemouth                 6      1
Arsenal FC                      6      8
Aston Villa FC                  3      5
Brighton & Hove Albion FC       7      7
Burnley FC                      4      5
Chelsea FC                      3      3
Crystal Palace FC               5      5
Everton FC                      7      3
Leicester City FC               4      4
Liverpool FC                    1      2
Manchester City FC              2      1
Manchester United FC            7      5
Newcastle United FC             8      3
Norwich City FC                 3      3
Sheffield United FC             3      9
Southampton FC                  3      4
Tottenham Hotspur FC            3      8
Watford FC                      6      4
West Ham United FC              4      5
Wolverhampton Wanderers FC      7      7

In [152]:
k=dframe.groupby(df['team1'])[['win1','win2']].sum()

In [157]:
k_new=k.rename({'win2':'loss1'},axis=1)
k_new

win1  loss1
team1                                  
AFC Bournemouth                5      8
Arsenal FC                    10      3
Aston Villa FC                 7      9
Brighton & Hove Albion FC      5      7
Burnley FC                     8      7
Chelsea FC                    11      5
Crystal Palace FC              6      8
Everton FC                     8      4
Leicester City FC             11      4
Liverpool FC                  18      0
Manchester City FC            15      2
Manchester United FC          10      2
Newcastle United FC            6      5
Norwich City FC                4     12
Sheffield United FC           10      6
Southampton FC                 6     10
Tottenham Hotspur FC          12      4
Watford FC                     6      7
West Ham United FC             6      9
Wolverhampton Wanderers FC     8      4

In [154]:
kk=dframe.groupby(df['team2'])[['win1','win2']].sum()

win1  win2
team2                                 
AFC Bournemouth               14     4
Arsenal FC                     7     4
Aston Villa FC                12     2
Brighton & Hove Albion FC      8     4
Burnley FC                     7     7
Chelsea FC                     7     9
Crystal Palace FC              9     5
Everton FC                    11     5
Leicester City FC              8     7
Liverpool FC                   3    14
Manchester City FC             7    11
Manchester United FC           6     8
Newcastle United FC           11     5
Norwich City FC               15     1
Sheffield United FC            6     4
Southampton FC                 6     9
Tottenham Hotspur FC           7     4
Watford FC                    13     2
West Ham United FC            10     4
Wolverhampton Wanderers FC     5     7

In [159]:
kk_new=kk.rename({'win1':'loss2'},axis=1)
kk_new

loss2  win2
team2                                  
AFC Bournemouth                14     4
Arsenal FC                      7     4
Aston Villa FC                 12     2
Brighton & Hove Albion FC       8     4
Burnley FC                      7     7
Chelsea FC                      7     9
Crystal Palace FC               9     5
Everton FC                     11     5
Leicester City FC               8     7
Liverpool FC                    3    14
Manchester City FC              7    11
Manchester United FC            6     8
Newcastle United FC            11     5
Norwich City FC                15     1
Sheffield United FC             6     4
Southampton FC                  6     9
Tottenham Hotspur FC            7     4
Watford FC                     13     2
West Ham United FC             10     4
Wolverhampton Wanderers FC      5     7

In [163]:
dd=pd.concat([k_new,kk_new],axis=1)
dd

win1  loss1  loss2  win2
AFC Bournemouth                5      8     14     4
Arsenal FC                    10      3      7     4
Aston Villa FC                 7      9     12     2
Brighton & Hove Albion FC      5      7      8     4
Burnley FC                     8      7      7     7
Chelsea FC                    11      5      7     9
Crystal Palace FC              6      8      9     5
Everton FC                     8      4     11     5
Leicester City FC             11      4      8     7
Liverpool FC                  18      0      3    14
Manchester City FC            15      2      7    11
Manchester United FC          10      2      6     8
Newcastle United FC            6      5     11     5
Norwich City FC                4     12     15     1
Sheffield United FC           10      6      6     4
Southampton FC                 6     10      6     9
Tottenham Hotspur FC          12      4      7     4
Watford FC                     6      7     13     2
West Ham United FC             6      9     10     4
Wolverhampton Wanderers FC     8      4      5     7

In [182]:
dw=pd.concat([dd,dr],axis=1)
dw

win1  loss1  loss2  win2  draw1  draw2
AFC Bournemouth                5      8     14     4      6      1
Arsenal FC                    10      3      7     4      6      8
Aston Villa FC                 7      9     12     2      3      5
Brighton & Hove Albion FC      5      7      8     4      7      7
Burnley FC                     8      7      7     7      4      5
Chelsea FC                    11      5      7     9      3      3
Crystal Palace FC              6      8      9     5      5      5
Everton FC                     8      4     11     5      7      3
Leicester City FC             11      4      8     7      4      4
Liverpool FC                  18      0      3    14      1      2
Manchester City FC            15      2      7    11      2      1
Manchester United FC          10      2      6     8      7      5
Newcastle United FC            6      5     11     5      8      3
Norwich City FC                4     12     15     1      3      3
Sheffield United FC           10      6      6     4      3      9
Southampton FC                 6     10      6     9      3      4
Tottenham Hotspur FC          12      4      7     4      3      8
Watford FC                     6      7     13     2      6      4
West Ham United FC             6      9     10     4      4      5
Wolverhampton Wanderers FC     8      4      5     7      7      7

In [183]:
dw['W']=dw['win1'] + dw['win2']
dw['L']=dw['loss1'] + dw['loss2']
dw['D']=dw['draw1'] + dw['draw2']
dw['Pts']= dw['W']*3 + dw['D']

In [184]:
dw

win1  loss1  loss2  win2  draw1  draw2   W   L  \
AFC Bournemouth                5      8     14     4      6      1   9  22   
Arsenal FC                    10      3      7     4      6      8  14  10   
Aston Villa FC                 7      9     12     2      3      5   9  21   
Brighton & Hove Albion FC      5      7      8     4      7      7   9  15   
Burnley FC                     8      7      7     7      4      5  15  14   
Chelsea FC                    11      5      7     9      3      3  20  12   
Crystal Palace FC              6      8      9     5      5      5  11  17   
Everton FC                     8      4     11     5      7      3  13  15   
Leicester City FC             11      4      8     7      4      4  18  12   
Liverpool FC                  18      0      3    14      1      2  32   3   
Manchester City FC            15      2      7    11      2      1  26   9   
Manchester United FC          10      2      6     8      7      5  18   8   
Newcastle United FC            6      5     11     5      8      3  11  16   
Norwich City FC                4     12     15     1      3      3   5  27   
Sheffield United FC           10      6      6     4      3      9  14  12   
Southampton FC                 6     10      6     9      3      4  15  16   
Tottenham Hotspur FC          12      4      7     4      3      8  16  11   
Watford FC                     6      7     13     2      6      4   8  20   
West Ham United FC             6      9     10     4      4      5  10  19   
Wolverhampton Wanderers FC     8      4      5     7      7      7  15   9   

                             D  Pts  
AFC Bournemouth              7   34  
Arsenal FC                  14   56  
Aston Villa FC               8   35  
Brighton & Hove Albion FC   14   41  
Burnley FC                   9   54  
Chelsea FC                   6   66  
Crystal Palace FC           10   43  
Everton FC                  10   49  
Leicester City FC            8   62  
Liverpool FC                 3   99  
Manchester City FC           3   81  
Manchester United FC        12   66  
Newcastle United FC         11   44  
Norwich City FC              6   21  
Sheffield United FC         12   54  
Southampton FC               7   52  
Tottenham Hotspur FC        11   59  
Watford FC                  10   34  
West Ham United FC           9   39  
Wolverhampton Wanderers FC  14   59

In [185]:
dw.sort_values(by='Pts', ascending=False)

win1  loss1  loss2  win2  draw1  draw2   W   L  \
Liverpool FC                  18      0      3    14      1      2  32   3   
Manchester City FC            15      2      7    11      2      1  26   9   
Manchester United FC          10      2      6     8      7      5  18   8   
Chelsea FC                    11      5      7     9      3      3  20  12   
Leicester City FC             11      4      8     7      4      4  18  12   
Tottenham Hotspur FC          12      4      7     4      3      8  16  11   
Wolverhampton Wanderers FC     8      4      5     7      7      7  15   9   
Arsenal FC                    10      3      7     4      6      8  14  10   
Burnley FC                     8      7      7     7      4      5  15  14   
Sheffield United FC           10      6      6     4      3      9  14  12   
Southampton FC                 6     10      6     9      3      4  15  16   
Everton FC                     8      4     11     5      7      3  13  15   
Newcastle United FC            6      5     11     5      8      3  11  16   
Crystal Palace FC              6      8      9     5      5      5  11  17   
Brighton & Hove Albion FC      5      7      8     4      7      7   9  15   
West Ham United FC             6      9     10     4      4      5  10  19   
Aston Villa FC                 7      9     12     2      3      5   9  21   
Watford FC                     6      7     13     2      6      4   8  20   
AFC Bournemouth                5      8     14     4      6      1   9  22   
Norwich City FC                4     12     15     1      3      3   5  27   

                             D  Pts  
Liverpool FC                 3   99  
Manchester City FC           3   81  
Manchester United FC        12   66  
Chelsea FC                   6   66  
Leicester City FC            8   62  
Tottenham Hotspur FC        11   59  
Wolverhampton Wanderers FC  14   59  
Arsenal FC                  14   56  
Burnley FC                   9   54  
Sheffield United FC         12   54  
Southampton FC               7   52  
Everton FC                  10   49  
Newcastle United FC         11   44  
Crystal Palace FC           10   43  
Brighton & Hove Albion FC   14   41  
West Ham United FC           9   39  
Aston Villa FC               8   35  
Watford FC                  10   34  
AFC Bournemouth              7   34  
Norwich City FC              6   21

### Transforming the request into a dataframe

In [7]:
match_date=[]
team_1=[]
team_2=[]
ft_score1=[]
ft_score2=[]
for m in data["matches"]:
    match_date.append(m["date"])
    team_1.append(m["team1"])
    team_2.append(m["team2"])
    ft_score1.append(m["score"]["ft"][0])
    ft_score2.append(m["score"]["ft"][1])
    

In [8]:
import pandas as pd
df=pd.DataFrame({'date':match_date,'team1':team_1,'score1':ft_score1,'team2':team_2,'score2':ft_score2})

In [9]:
df

date                team1  score1                 team2  score2
0    2019-08-09         Liverpool FC       4       Norwich City FC       1
1    2019-08-10   West Ham United FC       0    Manchester City FC       5
2    2019-08-10           Burnley FC       3        Southampton FC       0
3    2019-08-10      AFC Bournemouth       1   Sheffield United FC       1
4    2019-08-10    Crystal Palace FC       0            Everton FC       0
..          ...                  ...     ...                   ...     ...
375  2020-07-26    Leicester City FC       0  Manchester United FC       2
376  2020-07-26   Manchester City FC       5       Norwich City FC       0
377  2020-07-26  Newcastle United FC       1          Liverpool FC       3
378  2020-07-26       Southampton FC       3   Sheffield United FC       1
379  2020-07-26   West Ham United FC       1        Aston Villa FC       1

[380 rows x 5 columns]

### An EDA into the dataframe

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 380 entries, 0 to 379
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   date    380 non-null    object
 1   team1   380 non-null    object
 2   score1  380 non-null    int64 
 3   team2   380 non-null    object
 4   score2  380 non-null    int64 
dtypes: int64(2), object(3)
memory usage: 15.0+ KB


#### Transforming 'date' from object into datetime time-stamp

In [11]:
df["date"]=pd.to_datetime(df["date"],format="%Y/%m/%d")
df

date                team1  score1                 team2  score2
0   2019-08-09         Liverpool FC       4       Norwich City FC       1
1   2019-08-10   West Ham United FC       0    Manchester City FC       5
2   2019-08-10           Burnley FC       3        Southampton FC       0
3   2019-08-10      AFC Bournemouth       1   Sheffield United FC       1
4   2019-08-10    Crystal Palace FC       0            Everton FC       0
..         ...                  ...     ...                   ...     ...
375 2020-07-26    Leicester City FC       0  Manchester United FC       2
376 2020-07-26   Manchester City FC       5       Norwich City FC       0
377 2020-07-26  Newcastle United FC       1          Liverpool FC       3
378 2020-07-26       Southampton FC       3   Sheffield United FC       1
379 2020-07-26   West Ham United FC       1        Aston Villa FC       1

[380 rows x 5 columns]

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 380 entries, 0 to 379
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype         
---  ------  --------------  -----         
 0   date    380 non-null    datetime64[ns]
 1   team1   380 non-null    object        
 2   score1  380 non-null    int64         
 3   team2   380 non-null    object        
 4   score2  380 non-null    int64         
dtypes: datetime64[ns](1), int64(2), object(2)
memory usage: 15.0+ KB


<span style='color:purple'>'date' has been successfully transformed into datetime time-stamp and made ready for Time Series Analysis </span>

#### Contending teams and number of matches played

In [14]:
df.describe(include='object')

team1                 team2
count                    380                   380
unique                    20                    20
top     Tottenham Hotspur FC  Tottenham Hotspur FC
freq                      19                    19

In [14]:
df['team1'].nunique(), df['team2'].nunique()

(20, 20)

In [15]:
len(df)/20

19.0

<span style='color:purple'>20 teams participated and played a total of 380 matches with individual team playing 19 matches each at home and 19 matches away from home. The implication is that for the entire season each team plays a total of 38 matches (19 mulply by 2)</span>

#### Exploring goals scored

In [16]:
df.describe()

score1      score2
count  380.000000  380.000000
mean     1.515789    1.205263
std      1.248646    1.200252
min      0.000000    0.000000
25%      1.000000    0.000000
50%      1.000000    1.000000
75%      2.000000    2.000000
max      8.000000    9.000000

On avarage, more goals appears to be scored playing at home than playing away from home. However, the highest number of goals in a single match came while playing away from home

In [17]:
df.sort_values(by='score1', ascending=False)

date                 team1  score1                       team2  \
54  2019-09-21    Manchester City FC       8                  Watford FC   
147 2019-12-04          Liverpool FC       5                  Everton FC   
299 2020-06-22    Manchester City FC       5                  Burnley FC   
376 2020-07-26    Manchester City FC       5             Norwich City FC   
152 2019-12-07  Tottenham Hotspur FC       5                  Burnley FC   
..         ...                   ...     ...                         ...   
310 2020-06-27        Aston Villa FC       0  Wolverhampton Wanderers FC   
308 2020-06-25        Southampton FC       0                  Arsenal FC   
49  2019-09-16        Aston Villa FC       0          West Ham United FC   
151 2019-12-07       AFC Bournemouth       0                Liverpool FC   
53  2019-09-21            Everton FC       0         Sheffield United FC   

     score2  
54        0  
147       2  
299       0  
376       0  
152       0  
..      ...  
310       1  
308       2  
49        0  
151       3  
53        2  

[380 rows x 5 columns]

In [18]:
df.sort_values(by='score2', ascending=False)

date                       team1  score1                 team2  \
90  2019-10-25              Southampton FC       0     Leicester City FC   
218 2020-01-12              Aston Villa FC       1    Manchester City FC   
73  2019-10-05             Norwich City FC       1        Aston Villa FC   
1   2019-08-10          West Ham United FC       0    Manchester City FC   
45  2019-09-14  Wolverhampton Wanderers FC       2            Chelsea FC   
..         ...                         ...     ...                   ...   
273 2020-02-29         Newcastle United FC       0            Burnley FC   
100 2019-11-02             AFC Bournemouth       1  Manchester United FC   
275 2020-02-29                  Watford FC       3          Liverpool FC   
279 2020-03-07                  Arsenal FC       1    West Ham United FC   
190 2019-12-28                  Watford FC       3        Aston Villa FC   

     score2  
90        9  
218       6  
73        5  
1         5  
45        5  
..      ...  
273       0  
100       0  
275       0  
279       0  
190       0  

[380 rows x 5 columns]

The highest number of goals scored in a single match was 9 goals which appears to be an away goals by Leicester City against Souththampton in October 25, 2019 and the highest home goal scored was 8 goals by Manchester City against Watford on September 21, 2019.

In [21]:
home_goals=df['score1'].sum()
home_goals

576

In [22]:
away_goals=df['score2'].sum()
away_goals

458

In [23]:
total_goals=df['score1'].sum() + df['score2'].sum()
total_goals

1034

Total numbers of goals scored in the 2019/2020 premiership leagues is 1034. 576 of which are goals scored at home and 458 away goals

#### Wins, losses and draws

In [116]:
home_wins=df[(df['score1'] > df['score2'])]
home_wins

date                 team1  score1                       team2  \
0   2019-08-09          Liverpool FC       4             Norwich City FC   
2   2019-08-10            Burnley FC       3              Southampton FC   
6   2019-08-10  Tottenham Hotspur FC       3              Aston Villa FC   
9   2019-08-11  Manchester United FC       4                  Chelsea FC   
10  2019-08-17            Arsenal FC       2                  Burnley FC   
..         ...                   ...     ...                         ...   
369 2020-07-22          Liverpool FC       5                  Chelsea FC   
370 2020-07-26            Arsenal FC       3                  Watford FC   
372 2020-07-26            Chelsea FC       2  Wolverhampton Wanderers FC   
376 2020-07-26    Manchester City FC       5             Norwich City FC   
378 2020-07-26        Southampton FC       3         Sheffield United FC   

     score2  
0         1  
2         0  
6         1  
9         0  
10        1  
..      ...  
369       3  
370       2  
372       0  
376       0  
378       1  

[172 rows x 5 columns]

In [71]:
away_wins=df[(df['score2'] > df['score1'])]
away_wins

date                team1  score1                      team2  score2
1   2019-08-10   West Ham United FC       0         Manchester City FC       5
5   2019-08-10           Watford FC       0  Brighton & Hove Albion FC       3
8   2019-08-11  Newcastle United FC       0                 Arsenal FC       1
13  2019-08-17       Aston Villa FC       1            AFC Bournemouth       2
15  2019-08-17       Southampton FC       1               Liverpool FC       2
..         ...                  ...     ...                        ...     ...
366 2020-07-21           Watford FC       0         Manchester City FC       4
371 2020-07-26           Burnley FC       1  Brighton & Hove Albion FC       2
374 2020-07-26           Everton FC       1            AFC Bournemouth       3
375 2020-07-26    Leicester City FC       0       Manchester United FC       2
377 2020-07-26  Newcastle United FC       1               Liverpool FC       3

[116 rows x 5 columns]

There are 172 home wins as against 116 away victory in the 2019/2020 premiership season

In [41]:
df_draws=df[df['score1']==df['score2']]
df_draws.sort_values(by='score1', ascending=False)

date                       team1  score1                      team2  \
246 2020-02-01          West Ham United FC       3  Brighton & Hove Albion FC   
128 2019-11-24         Sheffield United FC       3       Manchester United FC   
139 2019-12-01        Manchester United FC       2             Aston Villa FC   
240 2020-02-01           Leicester City FC       2                 Chelsea FC   
234 2020-01-21                  Chelsea FC       2                 Arsenal FC   
..         ...                         ...     ...                        ...   
273 2020-02-29         Newcastle United FC       0                 Burnley FC   
244 2020-02-01         Newcastle United FC       0            Norwich City FC   
93  2019-10-26                  Watford FC       0            AFC Bournemouth   
253 2020-02-14  Wolverhampton Wanderers FC       0          Leicester City FC   
4   2019-08-10           Crystal Palace FC       0                 Everton FC   

     score2  
246       3  
128       3  
139       2  
240       2  
234       2  
..      ...  
273       0  
244       0  
93        0  
253       0  
4         0  

[92 rows x 5 columns]

In [20]:
len(df_draws)/len(df)

0.24210526315789474

92 matches ended in a draw which amount to only 24.2% (nearly a quarter) of the total matches played with the highest goals scored at 3 apiece 

#### Goals for and goals against

In [59]:
h=df.groupby(df['team1'])[['score1','score2']].sum()

In [71]:
h_new=h.rename({'score2':'conceed1'},axis=1)

In [73]:
h_new

score1  conceed1
team1                                       
AFC Bournemouth                 22        30
Arsenal FC                      36        24
Aston Villa FC                  22        30
Brighton & Hove Albion FC       20        27
Burnley FC                      24        23
Chelsea FC                      30        16
Crystal Palace FC               15        20
Everton FC                      24        21
Leicester City FC               35        17
Liverpool FC                    52        16
Manchester City FC              57        13
Manchester United FC            40        17
Newcastle United FC             20        21
Norwich City FC                 19        37
Sheffield United FC             24        15
Southampton FC                  21        35
Tottenham Hotspur FC            36        17
Watford FC                      22        27
West Ham United FC              30        33
Wolverhampton Wanderers FC      27        19

In [102]:
h_new.sort_values(by='conceed1',ascending=False)

score1  conceed1
team1                                       
Norwich City FC                 19        37
Southampton FC                  21        35
West Ham United FC              30        33
AFC Bournemouth                 22        30
Aston Villa FC                  22        30
Brighton & Hove Albion FC       20        27
Watford FC                      22        27
Arsenal FC                      36        24
Burnley FC                      24        23
Newcastle United FC             20        21
Everton FC                      24        21
Crystal Palace FC               15        20
Wolverhampton Wanderers FC      27        19
Manchester United FC            40        17
Tottenham Hotspur FC            36        17
Leicester City FC               35        17
Liverpool FC                    52        16
Chelsea FC                      30        16
Sheffield United FC             24        15
Manchester City FC              57        13

With teams playing at home, Norwich City conceeded the highest number of goals (i.e. 37 goals) while Manchester City conceeded the least with 13 goals against

In [103]:
p=df.groupby(df['team2'])[['score1','score2']].sum()
p

score1  score2
team2                                     
AFC Bournemouth                 35      18
Arsenal FC                      24      20
Aston Villa FC                  37      19
Brighton & Hove Albion FC       27      19
Burnley FC                      27      19
Chelsea FC                      38      39
Crystal Palace FC               30      16
Everton FC                      35      20
Leicester City FC               24      32
Liverpool FC                    17      33
Manchester City FC              22      45
Manchester United FC            19      26
Newcastle United FC             37      18
Norwich City FC                 38       7
Sheffield United FC             24      15
Southampton FC                  25      30
Tottenham Hotspur FC            30      25
Watford FC                      37      14
West Ham United FC              29      19
Wolverhampton Wanderers FC      21      24

In [104]:
p_new=p.rename({'score1':'conceed2'},axis=1)

In [105]:
p_new

conceed2  score2
team2                                       
AFC Bournemouth                   35      18
Arsenal FC                        24      20
Aston Villa FC                    37      19
Brighton & Hove Albion FC         27      19
Burnley FC                        27      19
Chelsea FC                        38      39
Crystal Palace FC                 30      16
Everton FC                        35      20
Leicester City FC                 24      32
Liverpool FC                      17      33
Manchester City FC                22      45
Manchester United FC              19      26
Newcastle United FC               37      18
Norwich City FC                   38       7
Sheffield United FC               24      15
Southampton FC                    25      30
Tottenham Hotspur FC              30      25
Watford FC                        37      14
West Ham United FC                29      19
Wolverhampton Wanderers FC        21      24

In [109]:
p_new.sort_values(by='conceed2', ascending=False)

conceed2  score2
team2                                       
Chelsea FC                        38      39
Norwich City FC                   38       7
Aston Villa FC                    37      19
Watford FC                        37      14
Newcastle United FC               37      18
AFC Bournemouth                   35      18
Everton FC                        35      20
Tottenham Hotspur FC              30      25
Crystal Palace FC                 30      16
West Ham United FC                29      19
Brighton & Hove Albion FC         27      19
Burnley FC                        27      19
Southampton FC                    25      30
Leicester City FC                 24      32
Arsenal FC                        24      20
Sheffield United FC               24      15
Manchester City FC                22      45
Wolverhampton Wanderers FC        21      24
Manchester United FC              19      26
Liverpool FC                      17      33

With teams playing away from home, Chelsea conceeded the highest number of goals(i.e. 38 goals) while Liverpool conceeded the least with 17 goals against

In [77]:
d=pd.concat([h_new,p_new],axis=1)
d

score1  conceed1  conceed2  score2
AFC Bournemouth                 22        30        35      18
Arsenal FC                      36        24        24      20
Aston Villa FC                  22        30        37      19
Brighton & Hove Albion FC       20        27        27      19
Burnley FC                      24        23        27      19
Chelsea FC                      30        16        38      39
Crystal Palace FC               15        20        30      16
Everton FC                      24        21        35      20
Leicester City FC               35        17        24      32
Liverpool FC                    52        16        17      33
Manchester City FC              57        13        22      45
Manchester United FC            40        17        19      26
Newcastle United FC             20        21        37      18
Norwich City FC                 19        37        38       7
Sheffield United FC             24        15        24      15
Southampton FC                  21        35        25      30
Tottenham Hotspur FC            36        17        30      25
Watford FC                      22        27        37      14
West Ham United FC              30        33        29      19
Wolverhampton Wanderers FC      27        19        21      24

#### Advancing

In [84]:
d[d['score1'].values < d['score2'].values]

score1  conceed1  conceed2  score2
Chelsea FC             30        16        38      39
Crystal Palace FC      15        20        30      16
Southampton FC         21        35        25      30

Chelsea topped the three teams that had more goals scored away than at home

In [85]:
d[d['conceed1'].values > d['conceed2'].values]

score1  conceed1  conceed2  score2
Southampton FC          21        35        25      30
West Ham United FC      30        33        29      19

Southampton emerges as the only team that had more goals scored and conceeded less playing away from home than at home

In [96]:
d['GF']=d['score1'].values + d['score2'].values
d['GA']=d['conceed1'].values + d['conceed2'].values
d['GD']=d['GF'] - d['GA']

In [97]:
d

score1  conceed1  conceed2  score2   GF  GA  GD
AFC Bournemouth                 22        30        35      18   40  65 -25
Arsenal FC                      36        24        24      20   56  48   8
Aston Villa FC                  22        30        37      19   41  67 -26
Brighton & Hove Albion FC       20        27        27      19   39  54 -15
Burnley FC                      24        23        27      19   43  50  -7
Chelsea FC                      30        16        38      39   69  54  15
Crystal Palace FC               15        20        30      16   31  50 -19
Everton FC                      24        21        35      20   44  56 -12
Leicester City FC               35        17        24      32   67  41  26
Liverpool FC                    52        16        17      33   85  33  52
Manchester City FC              57        13        22      45  102  35  67
Manchester United FC            40        17        19      26   66  36  30
Newcastle United FC             20        21        37      18   38  58 -20
Norwich City FC                 19        37        38       7   26  75 -49
Sheffield United FC             24        15        24      15   39  39   0
Southampton FC                  21        35        25      30   51  60  -9
Tottenham Hotspur FC            36        17        30      25   61  47  14
Watford FC                      22        27        37      14   36  64 -28
West Ham United FC              30        33        29      19   49  62 -13
Wolverhampton Wanderers FC      27        19        21      24   51  40  11

#### Premiership League 2019/2020 season top five contenders

In [114]:
d.sort_values(by= 'GF',ascending=False).head()

score1  conceed1  conceed2  score2   GF  GA  GD
Manchester City FC        57        13        22      45  102  35  67
Liverpool FC              52        16        17      33   85  33  52
Chelsea FC                30        16        38      39   69  54  15
Leicester City FC         35        17        24      32   67  41  26
Manchester United FC      40        17        19      26   66  36  30

Manchester City topped the goal chart with a total number of 102 goals in the season follwed by Liverpool, Chelsea Leicester City and Manchester United in that order. This reveals the top five contenders for the league season

#### Premiership League 2019/2020 season last five contenders

In [113]:
d.sort_values(by= 'GF',ascending=False).tail()

score1  conceed1  conceed2  score2  GF  GA  GD
Brighton & Hove Albion FC      20        27        27      19  39  54 -15
Newcastle United FC            20        21        37      18  38  58 -20
Watford FC                     22        27        37      14  36  64 -28
Crystal Palace FC              15        20        30      16  31  50 -19
Norwich City FC                19        37        38       7  26  75 -49

Norwich City came last with the least goals scored in the season

In [189]:
dx=d.loc[:,'GF':'GD']

In [192]:
dy=dw.loc[:,'W':'Pts']

In [195]:
pd.concat([dx,dy],axis=1).sort_values(by='Pts',ascending=False)

GF  GA  GD   W   L   D  Pts
Liverpool FC                 85  33  52  32   3   3   99
Manchester City FC          102  35  67  26   9   3   81
Manchester United FC         66  36  30  18   8  12   66
Chelsea FC                   69  54  15  20  12   6   66
Leicester City FC            67  41  26  18  12   8   62
Tottenham Hotspur FC         61  47  14  16  11  11   59
Wolverhampton Wanderers FC   51  40  11  15   9  14   59
Arsenal FC                   56  48   8  14  10  14   56
Burnley FC                   43  50  -7  15  14   9   54
Sheffield United FC          39  39   0  14  12  12   54
Southampton FC               51  60  -9  15  16   7   52
Everton FC                   44  56 -12  13  15  10   49
Newcastle United FC          38  58 -20  11  16  11   44
Crystal Palace FC            31  50 -19  11  17  10   43
Brighton & Hove Albion FC    39  54 -15   9  15  14   41
West Ham United FC           49  62 -13  10  19   9   39
Aston Villa FC               41  67 -26   9  21   8   35
Watford FC                   36  64 -28   8  20  10   34
AFC Bournemouth              40  65 -25   9  22   7   34
Norwich City FC              26  75 -49   5  27   6   21

This final table may be compared with the one [here](https://www.google.com/search?q=2019%2F20+premier+league+table&rlz=1C1JZAP_enFI760FI760&oq=&aqs=chrome.3.35i39i362l6j46i39i175i199i362j35i39i362...8.66315574j0j15&sourceid=chrome&ie=UTF-8#sie=lg;/g/11fj6snmjm;2;/m/02_tc;st;fp;1;;)

### Conclusion
Most objectives have been achieved, conducting an EDA using Time Series Analysis may help unfold the days and hours in which the matches where played and possibly the time in which half the goals where scored. I considered the resulting tables visual enough to exclude visuals such as pie charts and bar charts.